src/test/java/Task3package/Vegetable.java

```diff
@@ -0,0 +1,57 @@
+package Task3package;
+
+import java.util.Objects;
+
+public class Vegetable extends Сhef {
+    private String Name;
+    private String Color;
+    private double Weight;
+    private double Calories;
+
+    public Vegetable(String Name, String Color, double Weight, int Calories) {
+        this.Name = Name;
+        this.Color = Color;
+        this.Weight = Weight;
+        this.Calories = Calories;
+    }
+
+    public String getName() {
+        return Name;
+    }
+
+    public String getColor() {
+        return Color;
+    }
+
+    public double getWeight() {
+        return Weight;
+    }
+
+    public double getCalories() {
+        return Calories;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Vegetable vegetable = (Vegetable) o;
+        return Double.compare(vegetable.getWeight(), getWeight()) == 0 &&
+                Double.compare(vegetable.getCalories(), getCalories()) == 0 &&
+                Objects.equals(getName(), vegetable.getName()) &&
+                Objects.equals(getColor(), vegetable.getColor());
```

Comment:
After reading javadoc of Objects.equals method (java.lang.Object#equals) I found out this way of implementation that is stored under the hood:
`public static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }`
and
`public boolean equals(Object obj) {
        return (this == obj);
    }`

here we have `==` that is using for comparing primitives, while you're passing reference data type as a parameters. What would such a comparison bring you?

for more detailed explanation read this: [Equality](https://medium.com/better-programming/equality-and-comparison-in-java-pitfalls-and-best-practices-96b713e7009)